### PR TITLE
Add Fossil 2.7

### PR DESCRIPTION
**Pull request by bollwyvl**
Adds [fossil](https://www.fossil-scm.org) 2.7.

c/f #7418.

Local linux builds look fine, having some issues getting my local windows working properly, so it's off to CI :slot_machine: !

### CODE REVIEW

**Comment by bollwyvl** [comment]
Whew, that's a lot of fail!

[linux](https://circleci.com/gh/conda-forge/staged-recipes/34046?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)
```
+ make
mkdir bld
false -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -pipe -I$PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/fossil-2.7 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -o bld/translate ./src/translate.c
make: *** [src/main.mk:519: bld/translate] Error 1
```

[windows](https://ci.appveyor.com/project/conda-forge/staged-recipes/builds/21782483#L785):
```
Building OpenSSL from ..\compat\openssl-1.0.2p...
The system cannot find the path specified.
NMAKE : fatal error U1077: 'pushd' : return code '0x1'
```

**Comment by jakirkham** [comment]
Might need to enable verbosity (or turn up the volume) with `make` to get more information.

Is it possible to get `fossil` to link against our copy of `openssl` instead?

**Comment by bollwyvl** [comment]
Yeah I'll take another swing at it when I get a chance, but would be happy
for anyone else to give it a shot!

**Comment by bollwyvl** [comment]
It cracks me up how it just works on OSX... anyhow, it looks like it can't find the compiler on linux. Locally, it's falling back to my `cc`, which is why it's working. Trying a symlink...

**Comment by bollwyvl** [comment]
Huzzah, mostly :green_heart: !

@jakirkham any thoughts on the appveyor fail? And maybe how to avoid symlinking `cc`?

I also am seeing that the `msvc` build compiles in sqlite, without a concept of how to link against a "system" one. Are we concerned? It seems like going to the `mingw` chain would be heavy...

**Comment by jakirkham** [comment]
Is there a `FOSSIL_BUILD_SQLITE=0` that we need to be setting?

**Comment by bollwyvl** [comment]
here's the makefile in all it's glory:

https://www.fossil-scm.org/xfer/artifact?udc=1&ln=on&name=43f7676e773e04b7

the telling line is:
```
 # Link everything except SQLite dynamically?
```
The mingw one has the `USE_SYSTEM_SQLITE` of the unix makefiles. Not sure how it would feel about having
```
$(SRCDIR)\sqlite3.h \
```
patched out...

**Comment by bollwyvl** [comment]
At least this is the guidance on azure right now:
https://github.com/conda-forge/staged-recipes/issues/7715#issuecomment-460897996

Will have to raise my windows Makefile game to patch in building against CF sqlite...

File: recipes/fossil/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  home: https://fossil-scm.org
+  license: BSD-2-Clause
+  license_family: BSD
+  license_file: COPYRIGHT-BSD2.txt
+  summary: >
+    Fossil is a simple, high-reliability, distributed software configuration
+    management system with these advanced features:
+
+  doc_url: https://fossil-scm.org/index.html/doc/trunk/www/permutedindex.html
+  dev_url: https://github.com/simplejson/simplejson
```
Comment (jakirkham) [inline recipes/fossil/meta.yaml]: Just noticed this URL looked off. Have raised as issue ( https://github.com/conda-forge/fossil-feedstock/issues/4 ).

**Comment by jakirkham** [comment]
Thanks for working on this @bollwyvl! :)

**Pull request merged**